dll/appcompat/shims/genral/CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ add_library(acgenral SHARED
     ${SOURCE}
     ${CMAKE_CURRENT_BINARY_DIR}/acgenral.def)
 
-set_module_type(acgenral win32dll)
+set_module_type(acgenral win32dll ENTRYPOINT DllMain 12)
```

Comment:
This might be worth a comment. Why do you need to skip CRT initialization?

Reply:
The code in these modules is running very early at process creation, so it makes sense to minimize dependencies / initialization code.

Reply:
Definitely makes sense. I wonder if it would be possible to get rid of the msvcrt dependency entirely (in part, to make sure none of the stuff that requires DllMainCRTStartup is used)... not sure how much of it is used.

Reply:
I can remove the `msvcrt` dependency, and both msvc and gcc will fall back to ntdll functions then (`vsnprintf`, `vsnwprintf`, `memcmp`, `memset`)

Reply:
(even before that, there were no other crt functions used.